src/engine/controls/cuecontrol.cpp

```diff
@@ -44,6 +44,14 @@ inline mixxx::RgbColor::optional_t doubleToRgbColor(double value) {
     return mixxx::RgbColor::optional(colorCode);
 }
 
+inline int hotcueIndexToHotcueNumber(int hotcueIndex) {
+    return hotcueIndex + 1;
```

Comment:
DEBUG_ASSERT(hotcueIndex != Cue::kNoHotCue);

Reply:
This is valid.

Reply:
It is not. I suppose that if I change Cue::kNoHotCue from -1 to any other negative value like -2, -3, ... the code will fail. This reveals an invalid, implicit assumption about the actual value of the constant.

Reply:
No, it does not fail, because we also need the number 0 for the vergine state.

However it is a good idea to to assert reasonable vslues. So I have added:

DEBUG_ASSERT(hotcueIndex >= Cue::kNoHotCue);

Reply:
This is still incorrect. Only the comparison operators != and == are valid for Cue::kNoHotCue. This is supposed to be a singular, invalid value without any relation to valid values.

Reply:
OK, I have now two conditions, and forced push the last commit.

Reply:
Still wrong. Addition and subtraction is undefined for the special value Cue::kNoHotCue. It has to be treated like NaN.

Reply:
Ping

Reply:
I have now introduced kNoHotCueNumber and explicit handling for it, that allows to give it any value.